scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -557,29 +597,51 @@ def update_view_from_model(self, workspace_removed=None):
         else:
             self.selected_data = []
 
-    def check_workspaces_are_tf_asymmetry_compliant(self, workspace_list):
-        non_compliant_workspaces = [item for item in workspace_list if 'Group' not in item]
-        return False if non_compliant_workspaces else True
-
     def get_parameters_for_tf_function_calculation(self, fit_function):
         mode = 'Construct' if self.view.tf_asymmetry_mode else 'Extract'
-        workspace_list = [self.view.display_workspace] if self.view.fit_type == self.view.single_fit else self.selected_data
+        workspace_list = self.selected_data if self.view.is_simul_fit() else [
+            self.view.display_workspace]
         return {'InputFunction': fit_function,
                 'WorkspaceList': workspace_list,
                 'Mode': mode}
 
-    def create_multi_domain_function(self, function_list):
-        if not any(function_list):
-            return None
-        multi_domain_function = MultiDomainFunction()
-        for index, func in enumerate(function_list):
-            multi_domain_function.add(func)
-            multi_domain_function.setDomainIndex(index, index)
-
-        return multi_domain_function
-
     def _get_selected_groups_and_pairs(self):
         return self.context.group_pair_context.selected_groups + self.context.group_pair_context.selected_pairs
 
-    def _check_data_exists(self, guess_selection):
+    def _get_run_number_from_workspace(self, workspace_name):
```

Comment:
these should be the workspace naming file, as they are to do with naming. We have a method for getting the group or pair from the workspace name https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/ADSHandler/workspace_naming.py#L60